### DISCUSSION TITLE
Reddit API - New changes for Third Party Apps

### DISCUSSION THREAD

**Discussion by giwrgosmant** [Docile-Alligator/Infinity-For-Reddit#1440] [Q&A] [answered]
Reddit will begging to charging for access to its API for third party apps.
This is gonna affect Infinity right?

https://www.reddit.com/r/Android/comments/12rmi75/big_change_for_third_party_reddit_apps_as_reddit/

**Comment by Gitchick57** [comment]
They haven't responded since the news was released a few days ago. 



Apr 22, 2023, 12:01 AM by ***@***.***:

**Comment by blockhead-pippi** [comment]
https://arstechnica.com/gadgets/2023/05/reddits-api-pricing-results-in-shocking-20-million-a-year-bill-for-apollo/

https://libreddit.domain.glass/r/OutOfTheLoop/comments/13wwrzb/whats_going_on_with_reddit_phone_apps_having_to/

**Comment by Apacelus** [comment]
Would it be possible to use the mobile websites api, like newpipe for youtube does?

**Reply by DraconicNEO** [reply]
You mean web scraping as opposed to API use? I would say it would definitely be possible, although it would require a lot of work to get it working and the majority of Reddit app developers have been spoiled rotten by free API access, so they're probably not going to want to do it. At least not yet. 

Honestly in my opinion it's the most sustainable Way forward considering that Reddit is probably going to restrict API usage in general, even for individual users (they may not be doing that now but they definitely might in the future).

**Comment by lm41** [comment]
Is there anything new on this topic?

**Comment by Gitchick57** [comment]
Well, for me, I am not able to make comments right now and app is kind of buggy today.  And so it begins...



Jun 5, 2023, 12:56 PM by ***@***.***:

**Comment by blockhead-pippi** [comment]
https://www.pcgamer.com/major-subreddit-admins-are-going-to-war-with-reddit-over-monetization-changes-that-will-kill-many-third-party-apps/

Incomplete and Growing List of Participating Subreddits:

https://libreddit.domain.glass/r/ModCoord/comments/1401qw5/incomplete_and_growing_list_of_participating/

**Comment by ZolanaPrime** [comment]
Obviously the new pricing is ridiculous and really stupid from Reddit.  I did see that even with the changes, there's a "free" version of the API that allows 100 calls/minute if OAuth is used (unless I misunderstood).  Wouldn't it be possible for Infinity (or other app) users to just get their own API key, and keep using the app?  Not ideal, but it's better than using the janky crappy official app.

Assuming that'd work, would it be possible to add a new setting into Infinity where you can input your own API key as just another field on the login screen or something?

**Reply by Turhvjbufv** [reply]
You can if you fork the app and add that option but it's against reddit terms and service so Docile-Alligator wouldn't want to do that, but you can do it yourself if you know how

**Reply by DraconicNEO** [reply]
Note: Not illegal, against reddit's terms of service. There's a very big difference. Though Docile-Alligator is for some reason trying to continue relations with Reddit (hence why he's making Infinity paid) so it makes sense that he's not going to burn the bridge with them just yet.

**Reply by Turhvjbufv** [reply]
Right, my bad sorry

**Reply by aman207** [reply, marked as answer]
Here's a post describing how to build your own APK with your own API key. https://old.reddit.com/r/Infinity_For_Reddit/comments/14c2v5x/build_your_own_apk_with_your_personal_api_key_in/
